### PR TITLE
upgrade vendor and update files for K8s v1.26

### PR DESCRIPTION
**Pull request by kushagra98**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:
Update dependencies of VPA e2e tests for K8s v1.26. Updated the deprecated functions.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kushagra98** [comment]
@jbartosik can you please review this change. I have tested the changes.

File: vertical-pod-autoscaler/e2e/update-kubernetes-deps.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
```
Comment (jbartosik) [inline vertical-pod-autoscaler/e2e/update-kubernetes-deps.sh:1]: Why not use https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/hack/update-kubernetes-deps-in-e2e.sh ?

**Comment by kushagra98** [comment]
/hold
Updating vendor for VPA as well.

**Review by jbartosik — CHANGES_REQUESTED** [review]

File: vertical-pod-autoscaler/e2e/update-kubernetes-deps.sh (hunk 1 above)
Comment (kushagra98) [reply vertical-pod-autoscaler/e2e/update-kubernetes-deps.sh]: Yeah, I delete this file. It's a copy.

**Comment by kushagra98** [comment]
PR closed, too many conflicts with rebase. 
Reverting the last 5 changes -> rebase with new changes (changes in vendor) -> add my new changes on top of it -> remove conflicts -> create new PR.

**Pull request closed without merging**